src/Features/Core/Portable/Completion/Providers/AbstractSymbolCompletionProvider.cs

```diff
@@ -233,9 +234,18 @@ protected virtual Task<bool> IsSemanticTriggerCharacterAsync(Document document,
 
         private Task<ImmutableArray<ISymbol>> GetSymbolsWorker(int position, bool preselect, SyntaxContext context, OptionSet options, CancellationToken cancellationToken)
         {
-            return preselect
-                ? GetPreselectedSymbolsWorker(context, position, options, cancellationToken)
-                : GetSymbolsWorker(context, position, options, cancellationToken);
+            try
+            {
+                return preselect
+                    ? GetPreselectedSymbolsWorker(context, position, options, cancellationToken)
+                    : GetSymbolsWorker(context, position, options, cancellationToken);
+            }
+            catch (Exception e)
+            {
+                FatalError.Report(e);
```

Comment:
Instead of doing this, you should do it in an exception filter so that the stack isn't already unwound.

Also - should you be handling `OperationCanceledException` specially?